delivery_dropoff_site/models/sale_order.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2018 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    dropoff_site_required = fields.Boolean(
+        string="Drop-off Site Required",
+        store=True,
+        related="carrier_id.with_dropoff_site",
+    )
+
+    final_shipping_partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Final Recipient",
+        states={"draft": [("readonly", False)], "sent": [("readonly", False)]},
+        readonly=True,
+        help="It is the partner that will pick up the parcel " "in the dropoff site.",
+    )
+
+    @api.onchange("carrier_id")
+    def onchange_carrier_id(self):
+        carrier = self.carrier_id
+        partner = self.partner_shipping_id
+        if (
+            carrier
+            and partner
+            and (
+                carrier.with_dropoff_site != partner.is_dropoff_site
+                or (
+                    carrier.with_dropoff_site
+                    and partner.dropoff_site_carrier_id != carrier
+                )
+            )
+        ):
+            self.partner_shipping_id = False
+        if carrier and carrier.with_dropoff_site:
+            return {
```

Comment:
@jdoutreloux Returning domains is deprecated. Could you find another solution ?

Reply:
@rousseldenis fixed it. Thanks for the catch :)